src/main/java/bio/terra/service/snapshot/flight/export/SnapshotExportWriteManifestStep.java

```diff
@@ -65,25 +80,32 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
         GcsUriUtils.getGsPathFromComponents(
             exportBucket.getName(), String.format("%s/manifest.json", context.getFlightId()));
 
+    SnapshotModel snapshot = snapshotService.retrieveSnapshotModel(snapshotId, userReq);
+    Storage storage = gcsProjectFactory.getStorage(snapshot.getDataProject(), true);
     List<SnapshotExportResponseModelFormatParquetLocationTables> tables =
         paths.entrySet().stream()
             .map(
                 entry ->
                     new SnapshotExportResponseModelFormatParquetLocationTables()
                         .name(entry.getKey())
-                        .paths(entry.getValue()))
-            .collect(Collectors.toList());
-
-    SnapshotModel snapshot = snapshotService.retrieveSnapshotModel(snapshotId, userReq);
+                        .paths(
+                            signUrls
+                                ? entry.getValue().stream()
+                                    .map(path -> signGsUrl(storage, path))
+                                    .toList()
+                                : entry.getValue()))
```

Comment:
Not sure it's much better but you could move the `?` inside the inner `map`, to save a line of code.
```suggestion
                                entry.getValue().stream()
                                    .map(path -> signUrls ? signGsUrl(storage, path) : path)
                                    .toList()
```

Reply:
Yeah, I'm not sure which I like better.  Kinda 50/50 so I'll just leave